playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx

```diff
@@ -33,21 +34,29 @@ const Checkbox = (props: CheckboxProps) => {
     data = {},
     error = false,
     id,
+    indeterminate = false,
     name = '',
     onChange = () => {},
     tabIndex,
     text = '',
     value = '',
   } = props
 
+  const checkRef = useRef()
   const dataProps = buildDataProps(data)
   const ariaProps = buildAriaProps(aria)
   const classes = classnames(
-    buildCss('pb_checkbox_kit', { checked, error }),
+    buildCss('pb_checkbox_kit', { checked, error, indeterminate }),
     globalProps(props),
     className
   )
 
+  useEffect(() => {
+    if (checkRef.current) {
+      checkRef.current.indeterminate = indeterminate
```

Comment:
Indeterminate as attribute for a traditional DOM elements needs to be added onmount or rerender. Hence the `useEffect` & `useRef`.